tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -78,32 +80,50 @@ def sync(sync_ttl):
-                                # integration_id
-                                params = {
-                                    'TendrlContext.integration_id':
-                                    integration_id,
-                                }
-                                payload = {
-                                    "tags": [_target],
-                                    "run": _flow,
-                                    "status": "new",
-                                    "parameters": params,
-                                    "type": "node"
-                                }
-                                _job_id = str(uuid.uuid4())
-                                _job = Job(
-                                    job_id=_job_id,
-                                    status="new",
-                                    payload=payload,
-                                    timeout="no"
-                                ).save()
-                                while True:
-                                    _job = _job.load()
-                                    if _job.status in ["finished",
-                                                       "failed"]:
-                                        break
-
+                                # Raise an alert regarding the same
+                                msg = "New peers identified in cluster: %s. " \
+                                    "Make sure tendrl-ansible is executed for " \
+                                    "the new nodes so that expand cluster " \
+                                    "option can be triggered" % integration_id
+                                event_utils.emit_event(
+                                    "cluster_status",
+                                    "expand_pending",
+                                    msg,
+                                    "cluster_{0}".format(integration_id),
+                                    "WARNING",
+                                    integration_id=integration_id
+                                )
+                            integration_id = NS.tendrl_context.integration_id
+                            nodes_ids = json.loads(NS._int.client.read(
+                                "indexes/tags/tendrl/integration/%s" % integration_id
+                            ).value)
+                            cluster_peers = NS._int.client.read(
+                                "/clusters/%s/Peers" % integration_id
+                            )
+                            peer_count = 0
+                            for peer in cluster_peers.leaves:
+                                peer_count += 1
+                            _cluster = NS.tendrl.objects.Cluster(
+                                integration_id=integration_id
+                            ).load()
+                            if len(nodes_ids) == peer_count and \
+                                _cluster.status != "expand_pending":
```

Comment:
gluster status is always "expand_pending", because when we import cluster it always len(nodes_ids) == peer_count
Once this blcok is executed cluster status is marked as expand_pending in etcd, so when new node came for expand then no clearing alert raised

Reply:
Ah yes. So when new peers detected we need to set the cluster status as `new_peers_detected` say and when no of peers equals the no of nodes in indexes and cluster status is `new_peer_detected` then only we need to set it to `expand_pending`. Sent the PR with changes below.